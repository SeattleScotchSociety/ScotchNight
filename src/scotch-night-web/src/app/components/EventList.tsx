import { inject, observer } from "mobx-react";
import * as React from "react";
import { Button } from "reactstrap";

import { IEvent } from "../stores/EventStore";
import { IRootStore } from "../stores/RootStore";

import * as format from "date-fns/format";

interface IEventListProps { store: IRootStore; }

@inject("store")
@observer
export class EventList extends React.Component<IEventListProps> {
    constructor(props: IEventListProps) {
        super(props);
    }

    private onEventSelected(event: IEvent) {
        const { scotchNightStore, navigation } = this.props.store;

        scotchNightStore.setCurrentEvent(event);

        navigation.push(`/events/${event.id}`);
    }

    public render() {
        const { events } = this.props.store.eventStore;
        const eventElements = events.map((event) => {
            const { date, description, location, title, id } = event;

            return (
                <div className="box" key={id.toString()} onClick={() => this.onEventSelected(event)}>
                    <h3>{format(date, "MMMM D, YYYY")}</h3>
                    <p className="event__title">{title}</p>
                    <p className="event__desc">{description}</p>
                </div>
            );
        });

        return (
            <div className="container">
                <h1 className="page-header">Events</h1>
                <div className="events">
                    {eventElements}
                </div>
            </div>
        );
    }
}

export default EventList;
