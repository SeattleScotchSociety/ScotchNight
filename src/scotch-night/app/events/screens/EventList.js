// @flow
import React from 'react';
import { StyleSheet, Text, View, Image, FlatList, Button } from 'react-native';
import { connect } from 'react-redux';
import { getAllEvents } from '../api/EventsApi';
import EventListItem from '../components/EventListItem';

class EventList extends React.Component {
    static navigationOptions = {
        title: 'Events'
    };

    constructor(props) {
        super(props);
        const events = getAllEvents();
        this.state = {
            events
        };
    }

    componentWillMount() {
        const events = getAllEvents();

        this.setState({
            events
        });
    }

    _handleOnPress() {

    }

    _renderItem = ({ item }) => {
        return (
            <EventListItem
                event={item}
                onPress={() => navigate('EventDetail', { event: { ...item } })}
            />
        );
    };

    render() {
        return (
            <View style={styles.container}>
                <FlatList
                    data={this.state.events}
                    renderItem={this._renderItem}
                    keyExtractor={item => item.id}
                    ListHeaderComponent={this.renderHeader}
                    ListFooterComponent={this.renderFooter}
                />
                <Button title="Add New Event" color="#607D8B" onPress={this._handleOnPress} />
            </View>
        );
    }
}

function mapStateToProps() {
    return { events: [] };
}

export default connect(mapStateToProps)(EventList);

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    },
    headerText: {
        backgroundColor: 'transparent',
        color: 'white',
        fontSize: 24
    }
});
