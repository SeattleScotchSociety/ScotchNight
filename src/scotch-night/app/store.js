import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { AsyncStorage } from 'react-native';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer } from './Reducers';
import rootSaga from './Sagas';
import Bottles from './bottles';
let { loadBottles } = Bottles.Actions;

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
    reducer,
    undefined,
    compose(applyMiddleware(sagaMiddleware), autoRehydrate())
);

sagaMiddleware.run(rootSaga);

persistStore(store, { storage: AsyncStorage }, () => {
    store.dispatch(loadBottles());
}).purge();

export default store;
