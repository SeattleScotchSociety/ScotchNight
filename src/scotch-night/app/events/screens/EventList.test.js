import React from 'react';
import renderer from 'react-test-renderer';
import Store from '../../store';
import EventList from './EventList';

test('renders correctly', () => {
    const tree = renderer.create(<EventList store={Store} />).toJSON();
    expect(tree).toMatchSnapshot();
});
