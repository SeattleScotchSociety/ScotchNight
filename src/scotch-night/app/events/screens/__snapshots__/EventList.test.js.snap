// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignItems": "center",
        "height": 250,
        "justifyContent": "center",
        "width": null,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 24,
        }
      }
    >
      Seattle
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 24,
        }
      }
    >
      Scotch
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 24,
        }
      }
    >
      Society
    </Text>
  </Image>
  <ScrollView
    ListFooterComponent={undefined}
    ListHeaderComponent={undefined}
    data={
      Array [
        Object {
          "date": "5/30/2017",
          "description": "The first!",
          "id": 1,
          "location": "Phillabaums",
        },
        Object {
          "date": "7/2/2017",
          "description": "The second!",
          "id": 2,
          "location": "Fixs",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onViewableItemsChanged={undefined}
    removeClippedSubviews={true}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    shouldItemUpdate={[Function]}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View
      onLayout={[Function]}
    >
      <View
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          5/30/2017
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          7/2/2017
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
